src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -119,21 +120,50 @@ class Backend(object):
     """
     name = "openssl"
 
+    # FIPS doesn't allow several algorithms or weak key sizes, but the
+    # algorithms are still present in OpenSSL. They just errors if you try
+    # to use them. To avoid that we return from methods like hash_supported.
+    _non_fips_aead = {b'chacha20-poly1305'}
+    _non_fips_ciphers = (
+        ARC4, Blowfish, CAST5, Camellia, ChaCha20, IDEA, SEED
+    )
+    _non_fips_hashes = (hashes.MD5, hashes.BLAKE2b, hashes.BLAKE2s)
+    _rsa_min_key_size = 2048
+    _rsa_min_public_exponent = 65537
+    _dsa_min_modulus = 1 << 2048
+    _dh_min_key_size = 2048
+    _dh_min_modulus = 1 << _dh_min_key_size
+
     def __init__(self):
         self._binding = binding.Binding()
         self._ffi = self._binding.ffi
         self._lib = self._binding.lib
+        self._fips_enabled = self._is_fips_enabled()
 
         self._cipher_registry = {}
         self._register_default_ciphers()
-        self.activate_osrandom_engine()
+        if self._fips_enabled:
+            warnings.warn(
+                "OpenSSL FIPS mode is enabled. Can't enable DRBG fork safety.",
+                UserWarning
+            )
```

Comment:
a) This is wrong, we should only do this warning in cases where we're actually going to be use that engine (e.g. openssl<1.1.1d)
b) Should we even allow FIPS on OpenSSL's that have a broken internal CSPRNG? That's literally enabling a vulnerability.

Reply:
We absolutely must. There exist **no FIPS** openssl releases/validated distributions with the reseeding PRNG

Reply:
So there is no way to safely use a FIPS OpenSSL if your program forks?

Remind me why we're putting any work at all into this? I've worked for the US Government before, and I gotta say, if you need FIPS, maybe you should fucking do the work yourself instead of making volunteer OSS maintainers deal with NIST's braindamage.